### ISSUE TITLE
Integrate typescript-eslint v2 as soon as becomes stable

### ISSUE THREAD

**Issue by IlCallo** [quasarframework/app-extension-typescript#23]
https://github.com/typescript-eslint/typescript-eslint/issues/501#issuecomment-513580519

typescript-eslint v2 will be released as stable in a couple of weeks and contains some fixes for problems we noticed while moving away from TSLint

It must be updated after the release

**Comment by IlCallo** [comment]
Part of the update should be check new rules defaults (with or without Prettier)

**Comment by IlCallo** [comment]
V2 has been released.
There are some problems with Vue template linting, when it comes to JS/TS related directives, which I'm going to link here.

https://github.com/mysticatea/vue-eslint-parser/issues/55
https://github.com/vuejs/eslint-plugin-vue/issues/944
https://github.com/typescript-eslint/typescript-eslint/issues/883

Problems root cause is still under investigation.

**Comment by IlCallo** [comment]
Steps to execute the upgrade.

**.eslintrc.js**
`'plugin:@typescript-eslint/recommended-requiring-type-checking'` should be added right after `'plugin:@typescript-eslint/recommended'`, into `extends` array.

`extraFileExtensions: ['.vue']` should be added into `parserOptions`

Comments about `typescript-eslint` v2 release into `rules` property can be removed, as well as the `@typescript-eslint/indent` rule (style rules have been removed from recommended config).

**package.json**
`@typescript-eslint/eslint-plugin` and `"@typescript-eslint/parser` should be updated to latest release

**Fix linting problems**
_async usage where not needed_
`async` keywork should be removed from all boot files default exports, because it isn't actually used (there is no `await` inside any of those AFAIK)

_Vuex boilerplate_
`no-empty-function` complains about `actions`, `getters` and `mutations` files because they have an empty function body.
Adding a simple comment like `// your code` will solve this.
Unusued params will error too, I wrapped those into a multiline comment.

```ts
export function someAction(/* context */) {
  // your code
}
```

**Comment by IlCallo** [comment]
Update: https://github.com/typescript-eslint/typescript-eslint/issues/890
Someone is hopefully taking care of the problem

**Comment by IlCallo** [comment]
> There are some problems with Vue template linting

Problems have been solved (https://github.com/typescript-eslint/typescript-eslint/pull/1083) since v2.5.0

**Comment by IlCallo** [comment]
Fixed by #48
